src/rust/inetstack/protocols/tcp/established/sender.rs

```diff
@@ -314,31 +317,36 @@ impl Sender {
         }
     }
 
-    pub fn pop_one_unsent_byte(&self) -> Option<Buffer> {
+    pub fn pop_one_unsent_byte(&self) -> Option<DemiBuffer> {
         let mut queue = self.unsent_queue.borrow_mut();
 
         let buf = queue.front_mut()?;
         let mut cloned_buf = buf.clone();
         let buf_len: usize = buf.len();
 
         // Pop one byte off the buf still in the queue and all but one of the bytes on our clone.
-        buf.adjust(1);
-        cloned_buf.trim(buf_len - 1);
+        buf.adjust(1).expect("'buf' should contain at least one byte");
+        cloned_buf
+            .trim(buf_len - 1)
+            .expect("'cloned_buf' should contain at least one less than its professed length");
```

Comment:
Just curious, did you mean to use "professed" here? Or "processed"? - Both may be valid based on the context, but I may be missing the context here.

Reply:
I did mean "professed".  I was just being cheeky, I suppose I could have just used "claimed", but this wording was more fun.